marine/marine.py

```diff
@@ -236,3 +240,14 @@ def _collapse_macros(
             collapsed_result[field] = next(filter(None, possible_values), "")
 
         return collapsed_result
+
+    def _detect_encap(self, fields: List[str]) -> int:
```

Comment:
What's the performance hit here? I feel that if someone passes a lot of fields here, we're doing a lot of non-parsing work

Reply:
It appears like its around 5% slow down.

Reply:
Will a cache improve this? 5% is a lot of non-parsing work.
I'm thinking of a `fields -> encapsulation type` cache. What do you think?